lib/facility.rb

```diff
@@ -1,14 +1,68 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    if services.include?('Vehicle Registration')
+      @registered_vehicles << vehicle
+      vehicle.registration_date = Date.today
+      fees_and_plates(vehicle)
```

Comment:
I like that you've pulled this out to its own method instead of putting all the logic inside `register_vehicle`. That makes it more dynamic later on if fee change, for example.